__tests__/chat.test.js

```diff
@@ -219,46 +222,67 @@ describe('Chat Methods', () => {
           BOB_IS_SATISFIED = true
         }
       })
-      await alice.chat.send(teamChannel, {body: 'hello bob'})
-      await bob.chat.send(teamChannel, {body: 'hello alice'})
+      await alice1.chat.send(teamChannel, {body: 'hello bob'})
+      await bob.chat.send(teamChannel, {body: 'hello alice1'})
 
       await timeout(3000)
       expect(ALICE_IS_SATISFIED).toBe(true)
       expect(BOB_IS_SATISFIED).toBe(true)
     })
+
+    it("Doesn't pick up its own messages from the same device", async () => {
+      const messageText = 'Ever thus to deadbeats, Lebowski'
+      let noticedMessages = 0
+      alice1.chat.watchChannelForNewMessages(teamChannel, message => {
+        if (message.content.type === 'text' && message.content.text.body === messageText) {
+          noticedMessages++
+        }
+      })
+      await alice1.chat.send(teamChannel, {body: messageText})
+      await timeout(3000)
+      expect(noticedMessages).toBe(0)
+    })
   })
 
-  describe('watchAllChannelsForNewMessages', () => {
-    it('Can have bots count up to 10 to each other', async () => {
-      const STOP_AT = 10
-      const CONVO_CODE = crypto.randomBytes(8).toString('hex')
-      let HIGHEST_REACHED = 0
-
-      const onMessageForBot = (botName, bot) => {
-        const onMessage = async message => {
-          if (message.content.type === 'text') {
-            const body = message.content.text.body
-            if (body.indexOf(CONVO_CODE) !== -1) {
-              const num = parseInt(body.replace(CONVO_CODE, '').trim())
-              HIGHEST_REACHED = Math.max(num, HIGHEST_REACHED)
-              if (num < STOP_AT) {
-                const reply = {body: `${CONVO_CODE} ${num + 1}`}
-                await bot.chat.send(message.channel, reply)
-              }
+  const testTwoBotsCounting = async (bot1, bot2) => {
```

Comment:
I think it makes more sense to put this in the describe block for `watchAllChannelsForNewMessages`

Reply:
cool yeah, moving it in there.